docs/src/architecture/transactions.md

```diff
@@ -17,7 +17,7 @@ $
 It looks like 
 
 <p align="center">
-    <img src="../diagrams/protocol/transaction/Transaction.png">
+    <img src="../diagrams/architecture/transaction/Transaction.png">
 </p>
 
 In addition to specifying inputs and outputs, a transaction must also include a single executable script. This script has a well-defined structure which must do the following:
```

Comment:
The transaction script is optional and defined by the user, see [here](https://github.com/0xPolygonMiden/miden-base/blob/main/objects/src/transaction/compiled_tx.rs#L15).  I think the description you have described here is of the transaction program.

Reply:
Do you have more info on the transaction program and the transaction script? In which case would I use the transaction script?

This is what I have now

```
In addition to specifying inputs and outputs, a transaction must include an executable program. The transaction program has a well-defined structure which must do the following:

1. Build a single unified vault for all inputs.
2. Execute scripts of all input notes (the scripts are executed one after another).
3. Execute an optional user-defined script.
4. Build a single unified vault for all outputs.
5. Make sure the unified input and output vaults contain the same set of assets.

The last point ensures that a transaction does not create or destroy any assets.

A transaction can also include a user-defined transaction script. <Some more info here is missing>
```